examples/build_overrides/build.gni

```diff
@@ -15,5 +15,4 @@
 declare_args() {
   # Root directory for build files.
   build_root = "//third_party/connectedhomeip/build"
-  build_java_matter_controller = false
```

Comment:
The previous logic is "Set this flag when we build java-matter-controller", the current logic is "Set this flag when JAVA_PATH is defined". It looks to me the previous logic is more robust since environment variable could be set from any place

Reply:
agree, just move it back, thanks